gateway/handler_error.go

```diff
@@ -335,7 +335,7 @@ func (e *ErrorHandler) HandleError(w http.ResponseWriter, r *http.Request, errMs
 			NormalisePath(&record, &e.Spec.GlobalConfig)
 		}
 
-		if e.Spec.AnalyticsPlugin.Enabled {
+		if e.Spec.AnalyticsPluginConfig != nil {
```

Comment:
can we value that field `enabled`, and also in `handler_success.go:249`
```suggestion
		if e.Spec.AnalyticsPluginConfig.Enabled && e.Spec.AnalyticsPluginConfig != nil {
```

Reply:
but it is just not `nil` when it is enabled.

Reply:
it may not be nil, and still be disabled... like the case for `customDomain`

Reply:
I applied another solution, this is not an issue anymore.